Assignment2/Baseballhw.sql

```diff
@@ -0,0 +1,119 @@
+-- Show all the available databases
+SHOW DATABASES;
+
+-- Using the baseball database
+USE baseball;
+
+-- Show all the tables in the baseball database
+SHOW TABLES;
+
+-- Show game table
+
+SELECT *
+     FROM game
+     LIMIT 0, 20;
+
+-- Show batter_counts table
+
+SELECT *
+     FROM batter_counts
+     LIMIT 0,20;
+
+-- Adding index to game table
+ALTER TABLE game ADD INDEX `Gameid_date_ind`(`game_id`);
+
+-- Adding index to batter_counts table
+ALTER TABLE batter_counts ADD INDEX `Gameid_date_ind`(`game_id`);
+
+-- Creating a table to store batter stats
+DROP TABLE IF EXISTS s_batter_stats
+CREATE TABLE s_batter_stats
+    (INDEX Gameid_ind(game_id), INDEX Localdate(local_date), INDEX year_ind(Year))
+             SELECT baseball.game.game_id as game_id,
+                    DATE(baseball.game.local_date) as local_date,
+                    EXTRACT(YEAR FROM baseball.game.local_date) AS Year ,
```

Comment:
line 34: Use can use the YEAR() function directly to get the year from local_date.

`YEAR(local_date)`

Reply:
done that.